src/store/plugins/click-on-map-management.plugin.js

```diff
@@ -59,25 +55,45 @@ const clickOnMapManagementPlugin = (store) => {
         // when the user is not currently drawing something on the map
         if (mutation.type === 'setClickInfo' && !state.ui.showDrawingOverlay) {
             const clickInfo = mutation.payload
-            const isDesktopMode = store.getters.isDesktopMode
-            const isLeftClick = clickInfo?.clickType === ClickType.LEFT_CLICK
-            const isLongClick = clickInfo?.millisecondsSpentMouseDown >= 500
+            const isLeftSingleClick = clickInfo?.clickType === ClickType.LEFT_SINGLECLICK
+            const isContextMenuClick = clickInfo?.clickType === ClickType.CONTEXTMENU
+            const isFullscreenMode = store.state.ui.fullscreenMode
+
+            if (isLeftSingleClick) {
+                // Execute this before the then clause, as else the result could be wrong
+                // Still to do: why sometimes two clicks needed to fullscreen?
+                // Why crash when selecting search bar while popup is open?
+                const allowActivateFullscreen =
+                    !isFullscreenMode &&
+                    !state.features.selectedFeatures?.length &&
+                    !state.map.displayLocationPopup &&
+                    !state.search.show
 
-            if (
-                (isDesktopMode && isLeftClick) ||
-                (!isDesktopMode && isLeftClick && isLongClick) ||
-                (!isDesktopMode && !isLeftClick)
-            ) {
                 // if there are some search result shown, we hide the search list
                 if (state.search.show) {
                     store.dispatch('hideSearchResults')
                 }
                 // running an identification of feature even if we cleared the search result
-                runIdentify(store, clickInfo, store.getters.visibleLayers, store.state.i18n.lang)
-            } else if (!isDesktopMode && isLeftClick && !isLongClick) {
-                store.dispatch('toggleFullscreenMode')
-            } else if (isDesktopMode && !isLeftClick) {
+                runIdentify(
+                    store,
+                    clickInfo,
+                    store.getters.visibleLayers,
+                    store.state.i18n.lang
+                ).then((newSelectedFeatures) => {
+                    if (!newSelectedFeatures?.length && allowActivateFullscreen) {
+                        store.dispatch('toggleFullscreenMode')
+                    }
+                    store.dispatch('setSelectedFeatures', newSelectedFeatures)
+                })
+            }
+            if (isContextMenuClick) {
                 store.dispatch('clearAllSelectedFeatures')
+                store.commit('displayLocationPopup')
+            } else {
+                store.commit('hideLocationPopup')
```

Comment:
I would prefer that we stick to `store.dispatch`ing instead of using directly mutations, it is then easier to see what's happnening when debugging as we only have to watch actions to see what the app is doing (otherwise we have to go back and forth between actions and mutations to grasp what was happening).